### DISCUSSION TITLE
subclass of Response class should render open-api properly

### DISCUSSION THREAD

**Discussion by asomethings** [litestar-org/litestar#913] [Potential Issues]
I'm not sure if it's a bug or intended behavior so asking a question here.

When generating openapi schema it seems like [it only checks for StarliteResponse](https://github.com/starlite-api/starlite/blob/main/starlite/openapi/responses.py#L71) and  use wrong return annotation.
I think it should use `issubclass` to check class instead of using `is` operator. 

Below is a reproducible which will throw an `ImproperlyConfiguredException`. 
```python
from starlite import get
from starlite.response import Response as StarliteResponse

T = TypeVar("T")

class CustomResponse(StarliteResponse[T]):
    pass

@get()
def health() -> CustomResponse[str]:
    return CustomResponse('')
```

**Comment by Goldziher** [comment]
thanks, i created an issue for this - its indeed a bug. You're welcome to submit a PR fixing it if you have capacity.
